src/test/java/org/broadinstitute/hellbender/engine/filters/ReadFilterLibraryUnitTest.java

```diff
@@ -19,9 +18,170 @@
+    private SAMFileHeader getHeader() {
+        if (header == null) {
+            throw new IllegalArgumentException(
+                    "header has not been initialized;"
+                            + " beforeClass() was not invoked");
+        }
+        return header;
+    }
+
+    /**
+     * Composes the Id for the read group given its index.
+     *
+     * This methods must return a unique distinct ID for each possible index and
+     * it must be the same value each time it is invoked.
+     *
+     * @param index the index of the targeted read group in the range
+     *              [1, {@link #groupCount}]
+     * @return never <code>null</code> and must be unique to each possible
+     *         read group index.
+     */
+    private String composeReadGroupId(final int index) {
+        checkReadGroupIndex(index);
+        return DEFAULT_READ_GROUP_PREFIX + index;
+    }
+
+    /**
+     * Composes the Platform name for the read group given its index.
+     *
+     * This method must always return the same value give an index.
+     *
+     * @param index the index of the targeted read group in the range
+     *              [1, {@link #groupCount}]
+     * @return never <code>null</code>.
+     */
+    private String composePlatformName(final int index) {
+        checkReadGroupIndex(index);
+        return DEFAULT_PLATFORM_PREFIX + (((index-1)%2)+1);
+    }
+
+
+    /**
+     * Composes the Platform unit name for the read group given its index.
+     *
+     * @param index the index of the targeted read group in the range
+     *              [1,{@link #groupCount}]
+     * @return never <code>null</code>.
+     */
+    private String composePlatformUnitName(final int index) {
+        checkReadGroupIndex(index);
+        return DEFAULT_PLATFORM_UNIT_PREFIX + (((index-1)%3)+1);
+    }
+
+    /**
+     * Checks the correctness of a given read group index.
+     *
+     * A correct index is any value in the range [1,{@link #groupCount}].
+     *
+     * @param index the target index.
+     * @throws IllegalArgumentException if the input index is not correct.
+     */
+    private void checkReadGroupIndex(final int index) {
+        if (index < 1 || index > groupCount) {
+            throw new IllegalArgumentException(
+                    "read group" + " index ("
+                            + index
+                            + ") is out of bounds [1," + groupCount + "]");
+        }
+    }
+
+    /**
+     * Returns the ID for a read group given its index.
+     *
+     * @param index the index of the targeted read group in the range
+     *              [1,{@link #groupCount}]
+     * @return never <code>null</code> and must be unique to each
+     *              possible read group index.
+     */
+    private String getReadGroupId(final int index) {
+        checkReadGroupIndex(index);
+        return getHeader().getReadGroups().get(index - 1).getReadGroupId();
+    }
+
+    @BeforeClass
+    public void beforeClass() {
```

Comment:
It seems like this should be invoked directly by get header, or called in the constructor.